prov/sockets/src/sock_ep.c

```diff
@@ -1011,6 +1012,13 @@ static int sock_ep_getopt(fid_t fid, int level, int optname,
 		*optlen = sizeof(size_t);
 		break;
 
+	case FI_OPT_CM_DATA_SIZE:
+		if (*optlen < sizeof(size_t))
+			return -FI_ETOOSMALL;
```

Comment:
Yeah, that's a good point -- I note that the "succeed" code path does do exactly that (i.e., set `*optlen = sizeof(size_t)`).